charmstore.go

```diff
@@ -359,11 +359,21 @@ func (s *CharmStore) GetResource(id *charm.URL, revision int, name string) (resu
 	}, nil
 }
 
-// Publish tells the charmstore to mark the given charm as published with the given resource revisions.
-func (s *CharmStore) Publish(id *charm.URL, resources map[string]int) (*charm.URL, error) {
+// Publish tells the charmstore to mark the given charm as published with the
+// given resource revisions to the given channels. If no channels are specified,
+// the stable channel will be assumed.
+func (s *CharmStore) Publish(id *charm.URL, channels []string, resources map[string]int) (*charm.URL, error) {
```

Comment:
Why not just make this take a []params.Channel as an argument;
then there's no need to copy the slice.